pinot-core/src/main/java/org/apache/pinot/core/operator/filter/NullHandlingSupportedFilterOperator.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.operator.filter;
+
+import java.util.Arrays;
+import org.apache.pinot.core.common.BlockDocIdSet;
+import org.apache.pinot.core.operator.blocks.FilterBlock;
+import org.apache.pinot.core.operator.docidsets.AndDocIdSet;
+import org.apache.pinot.core.operator.docidsets.BitmapDocIdSet;
+import org.apache.pinot.core.operator.docidsets.MatchAllDocIdSet;
+import org.apache.pinot.core.query.request.context.QueryContext;
+import org.apache.pinot.segment.spi.datasource.DataSource;
+import org.apache.pinot.segment.spi.index.reader.NullValueVectorReader;
+
+
+public abstract class NullHandlingSupportedFilterOperator extends BaseFilterOperator {
```

Comment:
Do we need this extra layer of abstract class? In the end state all filter should have null handling and we can just put the new methods into `BaseFilterOperator`

Reply:
We can't put this logic inside `BaseFilterOperator` because some subclasses do not share the same logic. Here are some examples:

- Non-leaf filter operators, e.g. `AndFilterOperator`
- `TextMatchFilterOperator`
- `ExpressionFilterOperator` (the `nullBitmap` is from `TransformFunction::getNullBitmap` so we don't need `_dataSource`)

I renamed the class to `NullHandlingSupportedFilterOperator` to indicate it's the base class of a subset of filter operators.

We might do some tweaks to make some leaf filter operators a subclass of this class, but I'd defer that to not increase the scope of this PR.